ext/standard/basic_functions.c

```diff
@@ -3877,7 +3877,7 @@ PHP_FUNCTION(constant)
 	ZEND_PARSE_PARAMETERS_END();
 
 	scope = zend_get_executed_scope();
-	c = zend_get_constant_ex(const_name, scope, ZEND_FETCH_CLASS_SILENT);
+	c = zend_get_constant_ex(const_name, scope, ZEND_FETCH_CLASS_DEFAULT);
```

Comment:
This change causes test failures on CI. It looks like constant() doesn't want to see the usual error exceptions (e.g. for not found class) and instead only throws a constant not found warning. I'm wondering if we shouldn't just go with that behavior in this case as well? It's not super informative though...

Reply:
@nikic  The documentation states that const() will return null with an E_WARNING level error being generated if the constant is not defined so, my best guess, is to remove zend_throw_error call in zend_get_constant_ex when constant is undefined to match documentation. Also, added length check for class name because const("::") failed with fatal due to not finding class "" which is not the case.